src/Mocking.jl

```diff
@@ -50,6 +50,47 @@ function activate()
     return nothing
 end
 
+"""
+    Mocking.activate(f)
+
+
+Activate `@mock` call sites for the duration of the function `f`.
+
+!!! warning
+    This function redefines Mocking.jl internals and so can produce warnings about
+    method redefinitions. You may see the warning:
+        `WARNING: Method definition _activated($Int) at ... overwritten at ...`
+    This is expected behavior and can be safely ignored.
+    To avoid these warnings, instead use `Mocking.activate()` to activate `@mock` call
+    sites for the duration of the test suite. Alternatively, start Julia with
+    `--warn-overwrite=no` to suppress these warnings.
+"""
+function activate(f)
+    started_deactivated = !activated()
+    try
+        activate()
+        Base.invokelatest(f)
+    finally
+        started_deactivated && deactivate()
+    end
+end
+
+"""
+    Mocking.deactivate() -> Nothing
+
+Disable `@mock` call sites to only call the original function.
+
+!!! note
+    It is not usually necessary to call this function directly.
+    Instead it is recommended to simply call `Mocking.activate()` in `test/runtests.jl` to
+    activate `@mock` call sites for the duration of the test suite.
+"""
+function deactivate()
+    # Avoid redefining `_activated` when it's already set appropriately
+    Base.invokelatest(activated) && @eval _activated(::Int) = false
+    return nothing
+end
```

Comment:
Do you need `deactivate`? Unless you also require this for your code base it's probably best to avoid re-introducing this

Reply:
No, just for implementing (and testing) `activate(f)`, so perhaps we should delete the doc-string and name this `_deactivate()`?

Reply:
That would be good with me.